src/data_handling/timeseries.jl

```diff
@@ -48,6 +53,24 @@ function timeseries(notes, property, f, grid)
     return tvec, ts
 end
 
+"""
+    segment_notes(notes, grid)
+Quantizes the positions and durations of `notes` and segments them according to the duration of a grid unit.
+"""
+function segment_notes(notes, grid)
+    tpq = notes.tpq
+    segment_duration = Int(grid[2]*tpq)
+    grid_division = Int(tpq/segment_duration)
+    qnotes = quantize(notes, 0:1//grid_division:1)
```

Comment:
@johncwok Why are you making this choice? You are assuming that the user would only choose an equally spaced grid, but this is not the case always. for example `[0, 2//3, 1]` is a grid that we use regularly for our work.

Shouldn't this be modified to work with any grid? I think it is simple: For every note just find the grid point it belongs in (we already have code that does this) and then just add the same note for the subsequent grid-points that are covered by the note's duration.

Reply:
Oh you are right, I didn't take this into account.

Reply:
Okay but this actually makes sense though. If you want to use the segmented time-series for most of the traditional analyses methods, the spacing between two units must be equal. Nobody would ever use the `segment_notes` function with an irregular grid.

Reply:
> and for that the spacing needs to be equal.

Okay, yes, you have a good point. Let's leave it as is then, I'll add some documentation about the allowed grids then.